Profile page

Pytal opened the pull request:
### Tasks

- Profile info and actions
  - [x] Represent the info that is already set (DisplayName, Address, Avatar)
  - [x] Required Profile actions
    - [x] Account property actions (Email, Phone number, Website, Twitter)
    - [x] Apps can register actions
    - ~Additional email actions~
      - Order and scope of additional emails are not preserved, the logic for handling this is a prerequisite as visibility must also be preserved
  - [x] Profile manager
    - [x] Provide actions
    - [x] Provide profile info
  - [x] Privacy
    - [x] Profile visibility
    - [x] Respect already set scope settings
  - [x] Display user status
- Add new fields
  - [x] Enable Profile checkbox (enabled by default)
    - [x] Show error page when navigating to a disabled profile page
    - [x] Don't show profile entrypoints when disabled
  - [x] Organisation
  - [x] Role
  - [x] Biography
  - [x] Headline
- Profile entrypoints
  - [x] Add entrypoint to your own profile in top-right user menu
  - [x] Add profile preview card to personal info settings page
  - [x] Add profile entrypoint to contacts menu as action and displayname link
  - [x] Add profile entrypoint to Avatar menu through contacts menu actions provider (propagates to everything which uses ContactsMenuController)
- [x] Theming and dark mode work
- [x] Add necessary overrides (e.g. custom components, CSS overrides) for required functionality and UX
- [x] Make fields changes reactive (i.e. displayname, organisation, and profile enabled state changes are reflected instantly without needing a page reload)
- [x] Make it all look nice and work nicely together
- [x] Fix CI errors
- [ ] Cleanup

### Visual Reference
Profile page

![image](https://user-images.githubusercontent.com/24800714/137366168-117fd39c-e373-4066-acb5-695f04f7000f.png)

Settings page

![settings](https://user-images.githubusercontent.com/24800714/137365315-7e4c0aaf-a8d0-479a-afb9-a366f7808d47.png)

Contributes to https://github.com/nextcloud/server/issues/28139

ChristophWurst reviewed (changes requested):
Early feedback on the public API, hope you don't mind.

Public API is always to be taken with extra care. Once we define it, it can only be changed in non-breaking ways for the next three years.

Pytal committed:
Add profile templates

Signed-off-by: Christopher Ng <chrng8@gmail.com>

Pytal committed:
Add new account properties
- New properties
  - Organisation
  - Role
  - Headline
  - Biography
  - Profile Enabled property
- Fix errors with building default account properties
- Fix L10N factory method `getLanguage` not public error
- Update tests

Signed-off-by: Christopher Ng <chrng8@gmail.com>

Pytal committed:
Profile backend

Signed-off-by: Christopher Ng <chrng8@gmail.com>

Pytal committed:
Profile frontend

Signed-off-by: Christopher Ng <chrng8@gmail.com>
Signed-off-by: nextcloud-command <nextcloud-command@users.noreply.github.com>

PVince81 reviewed (commented):
Went through the whole codebase and only few minor things caught the eye, see comments.

Would be good to avoid reformatting whole files in the future as it only creates noise when revieweing in an already big PR.

ChristophWurst reviewed (approved):
Reviewed public API part and give it a smoke test

:+1: :shipit:
